HW2/P5/driver.py

```diff
@@ -77,6 +93,13 @@ def randcolor():
             print("{} simulation frames per second".format(frame_count / total_time))
             frame_count = 0
             total_time = 0
+
             # SUBPROBLEM 3: sort objects by location.  Be sure to update the
             # grid if objects' indices change!  Also be sure to sort the
             # velocities with their object positions!
+           
+            gridPos = (positions / grid_spacing).astype(int)
+            inds = np.argsort(map(cmp_zorder, gridPos, np.append(gridPos [1:], gridPos [:1], axis=0)))
```

Comment:
Ah, this looks much cleaner than mine. I really should have looked more into argsort...